lib/streams/join-stream.js

```diff
@@ -0,0 +1,33 @@
+var Transform = require('../transform');
+var inherits = require('util').inherits;
+
+/**
+ * JoinStream joins the incoming object stream into a data stream
+ * NOTE: writableObjectMode will always be true and readableObjectMode will always be false
+ *
+ * @class JoinStream
+ * @constructor
+ * @extends Transform
+ * @param {Object} [options] - Stream options object
+ */
+function JoinStream(options) {
```

Comment:
JoinStream is doing more than just interspersing a blank separator. 
It's also calling the `toString` method on all entires, and forcing the output into a data stream.

I'll revisit JoinStream once CompoundStreams are implemented